tests/marqo_test.py

```diff
@@ -106,12 +121,16 @@ def wrapper(self, *args, **kwargs):
 
 
 def create_settings_hash(settings_dict, kwargs):
-    combined_dict = {**settings_dict, **kwargs}
-    combined_str = ''.join(f"{key}{value}" for key, value in combined_dict.items())
+    """
+    Creates a hash from the settings dictionary and kwargs. Used to ensure that each index is created unique.
+    Size is restricted on 10 characters to prevent having to big index name which could cause issues.
+    """
+    dict_to_hash = {**settings_dict, **kwargs} if settings_dict else kwargs
```

Comment:
This may not be sufficient  to  ensure that `{"a": 1, "b":2 }` and `{"b":2, "a": 1}` convert to the same dict.

For example we could hash a serialised dict like here: https://stackoverflow.com/a/22003440

I'd recommend using tried-and-tested methods of serialising dicts than coming up with our own

Reply:
now that I think about it, when creating hash there shouldn’t be settings_dict and any kwargs passed simultaneously as settings_dict are prioritised over kwargs anyways. Instead changing it to `dict_to_hash = settings_dict if settings_dict else kwargs`

Reply:
yes you are correct - only 1 of <settings_dict> or <kwargs> makes sense.

But you haven't addressed my point. Two settings dicts, with the structure `{"a": 1, "b":2 }` and `{"b":2, "a": 1}` need to resolve to the same string, so that they can get the same hash. this is important to mitigate blowing up the number indexes that end up being made. 

One way we can enforce this is by sorting the keys. I suggest reading that link and using that as your serialisation procedure